### PR TITLE
[Merged by Bors] - chore: fix deprecation warnings

### PR DESCRIPTION
**Pull request by kim-em**
In preparation for https://github.com/leanprover/lean4/pull/3969 landing, we fix some deprecation warnings.

### CODE REVIEW

File: Mathlib/Algebra/GroupWithZero/Units/Basic.lean [hunk 1]
```diff
@@ -375,6 +375,7 @@ lemma div_eq_one_iff_eq (hb : b ≠ 0) : a / b = 1 ↔ a = b := hb.isUnit.div_eq
 lemma div_mul_cancel_right₀ (hb : b ≠ 0) (a : G₀) : b / (a * b) = a⁻¹ :=
   hb.isUnit.div_mul_cancel_right _
 
+set_option linter.deprecated false in
```
Comment (YaelDillies) [inline Mathlib/Algebra/GroupWithZero/Units/Basic.lean:378]: General remark, shouldn't the linter automatically be off on declarations that are themselves deprecated? cc @digama0

**Review by YaelDillies — APPROVED** [review]
maintainer merge

File: Mathlib/Algebra/GroupWithZero/Units/Basic.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Algebra/GroupWithZero/Units/Basic.lean:378]: I asked about this at https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/can.20the.20.60deprecated.60.20attribute.20set.20.60linter.2Edeprecated.20false.60/near/437754423

**Comment by semorrison** [comment]
bors p=10

**Comment by semorrison** [comment]
bors merge

File: Mathlib/Algebra/GroupWithZero/Units/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/GroupWithZero/Units/Basic.lean:378]: Nice! I hadn't seen this.

**Pull request closed without merging**